server-spi-private/src/main/java/org/keycloak/cookie/CookieType.java

```diff
@@ -20,7 +20,7 @@ public final class CookieType {
             .build();
 
     public static final CookieType AUTH_STATE = CookieType.create("KC_AUTH_STATE")
-            .scope(CookieScope.LEGACY_JS)
+            .scope(CookieScope.INTERNAL_JS)
```

Comment:
Need to make sure this doesn't need to be available after a redirect to Keycloak. I don't think so, but @mposolda can you confirm?

Reply:
This is set by KC right when handing the request, and it doesn't read it, and only used by the JS on login pages when in the same origin. So this shouldn't need to be SameSite=None

Reply:
@stianst This should be fine.

FYI. I am currently working on the change for "You are already logged-in" we discussed last week https://github.com/keycloak/keycloak/issues/27880 and it looks that this cookie will be possible to be removed entirely.